pkg/cloudprovider/gce/gce.go

```diff
@@ -115,7 +123,20 @@ func newGCECloud() (*GCECloud, error) {
 	if err != nil {
 		return nil, err
 	}
-	client := oauth2.NewClient(oauth2.NoContext, google.ComputeTokenSource(""))
+	tokenSource := google.ComputeTokenSource("")
+	if config != nil {
```

Comment:
Minor, but if you're going to split out some of the token source stuff into a separate file, this logic would seemingly belong there too.